irohad/model/converters/json_common.hpp

```diff
@@ -18,35 +18,336 @@
+         */
+        explicit FieldDeserializer(const D &document) : document(document) {}
+
+        /**
+         * Assign the value to the block member
+         * @tparam V - block member type
+         * @tparam B - block type
+         * @param block - block value for member assignment
+         * @param member - pointer to member in block
+         * @return block with deserialized member on success, nullopt otherwise
+         */
+        template <typename V, typename B>
+        auto assignObjectField(B block, V B::*member) {
+          return [=](auto transformed) mutable {
+            block.*member = transformed;
+            return nonstd::make_optional(block);
+          };
+        }
+
+        /**
+         * Assign the value to the block member. Block is wrapped in shared
+         * pointer
+         * @tparam V - block member type
+         * @tparam B - block type
+         * @param block - block value for member assignment
+         * @param member - pointer to member in block
+         * @return block with deserialized member on success, nullopt otherwise
+         */
+        template <typename V, typename B>
+        auto assignObjectField(std::shared_ptr<B> block, V B::*member) {
+          return [=](auto transformed) mutable {
+            (*block).*member = transformed;
+            return nonstd::make_optional(block);
+          };
+        }
+
+        /**
+         * Create function, which will deserialize document field with given
+         * verification and getter, transform the value to required type, and
+         * assign it to block member
+         * @tparam T - getter return type
+         * @tparam V - block member type
+         * @tparam B - block type
+         * @tparam Convert - transform function type
+         * @param member - pointer to member in block
+         * @param field - field name for deserialization
+         * @param transform - transform function from T to V
+         * @return function, which takes block, returns block with deserialized
+         * member on success, nullopt otherwise
+         */
+        template <typename T, typename V, typename B,
+                  typename Convert = Convert<V>>
+        auto deserialize(V B::*member, const std::string &field,
+                         Convert transform = Convert()) {
+          return [this, member, field, transform](auto block) {
+            return deserializeField<T>(document, field)
+                | transform
+                | this->assignObjectField(block, member);
+          };
+        }
+
+        /**
+         * Deserialize field of Uint type to given member field of block
+         * @tparam V - block member type
+         * @tparam B - block type
+         * @param member - pointer to member in block
+         * @param field - field name for deserialization
+         * @return @see deserialize
+         */
+        template <typename V, typename B>
+        auto Uint(V B::*member, const std::string &field) {
```

Comment:
The naming is a little bit confusing, can you change it so something more specific